backend/src/Altinn.AccessManagement.UI/Altinn.AccessManagement.UI.Core/Models/SingleRight/CheckDelegationAccess/CheckDelegationAccessDto/CheckDelegationAccessDto.cs

```diff
@@ -0,0 +1,31 @@
+using System.ComponentModel.DataAnnotations;
+
+namespace Altinn.AccessManagement.UI.Core.Models.SingleRight.CheckDelegationAccess.CheckDelegationAccessDto
+{
+    /// <summary>
+    ///     Checks whether the user can delegate the given right to the given party
+    /// </summary>
+    /// <param name="To">
+    ///     Used to identify the party the authenticated user is acting on behalf of./param>
+    ///     <param name="Resources">The delegation acccess check request object that's going to be consumed by the backend</param>
+    ///     <returns></returns>
+    public class CheckDelegationAccessDto
+    {
+        /// <summary>
+        ///     The To object used for delegation acccess check request object that's going to be consumed by the backend
+        /// </summary>
+        /// <param name="To">
+        ///     Used to identify the party the authenticated user is acting on behalf of./param>
+        ///     <returns></returns>
+        [Required]
+        public To To { get; set; }
+
+        /// <summary>
+        ///     The Resource object for delegation acccess check request object that's going to be consumed by the backend
+        /// </summary>
+        /// <param name="Resource">The delegation acccess check request object that's going to be consumed by the backend</param>
+        /// <returns></returns>
+        [Required]
+        public Resource Resource { get; set; }
```

Comment:
Any specific reason/need to specify individual classes for From/To/Resource?
Could in stead reuse the existing model for List<AttributeMatch> as used in the models for MaskinportenSchema, as all From/To/Resource properties are either a single or an array of AttributeMatch

Reply:
Yeah my thought it was a bit unneccessary to send in a list of the objects when that list in relation with UserAccessCheck will always contain only one object (but correct me if I'm wrong in that part) as frontend also would have to send in a list of objects but that list would always contain one object as far as I can see (if not I can change to use DelegationInput instead).

I think it was a bit clumsy to create a List of objects when creating objects of the type when in reality it can with that approach be created like (e.g. In tests):

CheckDelegationAccessDto dto = new CheckDelegationAccessDto
{
    To = new To("urn:altinn:organizationnumber", "123456789"),
    Resource = new Resource("urn:altinn:resource", "testapi")
};
```

Reply:
But after deleting the To field from the request object I'll change to use the object that's been made previously and delete CheckDelegationAccessDto